client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteSchemasAdminImpl.java

```diff
@@ -0,0 +1,386 @@
+        this.operationsFactory = operationsFactory;
+        this.operationDispatcher = operationDispatcher;
+        this.protostreamCache = cacheManager.<String, String>getCache(InternalCacheNames.PROTOBUF_METADATA_CACHE_NAME).withFlags(FORCE_RETURN_VALUE);
+        this.timeout = protostreamCache.getRemoteCacheContainer().getConfiguration().connectionTimeout();
+    }
+
+    @Override
+    public SchemaOpResult create(Schema schema) {
+        return create(schema.getName(), schema.getContent());
+    }
+
+    @Override
+    public Map<String, SchemaOpResult> create(FileDescriptorSource generatedSchema) {
+        Map<String, SchemaOpResult> results = new HashMap<>(generatedSchema.getFiles().size());
+        generatedSchema.getFiles().entrySet().forEach(entry -> {
+            results.put(entry.getKey(), create(entry.getKey(), entry.getValue()));
+        });
+
+        return results;
+    }
+
+    private SchemaOpResult create(String key, String schemaContent) {
+        var result = protostreamCache.putIfAbsent(key, schemaContent);
+        return result == null ? SchemaOpResult.DELETED : SchemaOpResult.NONE;
+    }
+
+    @Override
+    public CompletionStage<SchemaOpResult> createAsync(Schema schema) {
+        return protostreamCache.putIfAbsentAsync(schema.getName(), schema.getContent())
+                .thenApply(r -> r == null ? SchemaOpResult.CREATED : SchemaOpResult.NONE);
+    }
+
+    @Override
+    public SchemaOpResult update(Schema schema) {
+        return update(schema, false);
+    }
+
+    @Override
+    public SchemaOpResult update(Schema schema, boolean force) {
+        if (force) {
+            // replace without any check
+            String prevValue = protostreamCache.put(schema.getName(), schema.getContent());
+            return prevValue != null ? SchemaOpResult.UPDATED : SchemaOpResult.NONE;
+        }
+
+        var stored = protostreamCache.getWithMetadata(schema.getName());
+        boolean updateResult = false;
+        if (stored != null && !Objects.equals(stored.getValue(), schema.getContent())) {
+            if (protostreamCache.replaceWithVersion(schema.getName(), schema.getContent(), stored.getVersion())) {
+                updateResult = true;
+            }
+        }
+        return updateResult ? SchemaOpResult.UPDATED : SchemaOpResult.NONE;
+    }
+
+    @Override
+    public CompletionStage<SchemaOpResult> updateAsync(Schema schema) {
+        return updateAsync(schema, false);
+    }
+
+    @Override
+    public CompletionStage<SchemaOpResult> updateAsync(Schema schema, boolean force) {
+        if (force) {
+            // if force is true, replace without any check
+            return protostreamCache.putAsync(schema.getName(), schema.getContent())
+                    .thenApply(prevValue -> prevValue != null ? SchemaOpResult.UPDATED : SchemaOpResult.NONE)
+                    .exceptionally(ex -> {
+                        if (ex != null) {
```

Comment:
`ex` is always non null inside an `exceptionally()`
I don't think you need to wrap it in `InfinispanException` but if so, it is better to add a new constructor to receive a single throwable.